numpy/lib/tests/test_io.py

```diff
@@ -1823,6 +1828,123 @@ def test_auto_dtype_largeint(self):
         assert_equal(test['f1'], 17179869184)
         assert_equal(test['f2'], 1024)
 
+
+class TestPathUsage(TestCase):
+    @contextlib.contextmanager
```

Comment:
Ah, interesting!

I'm not sure I completely understood the problem, though.

From my understanding, a function like so:

```python
@contextlib.contextmanager
def with_file(fname):
    fd = open(fname)
    yield fd
    fd.close()
```

Will not actually run the `fd.close()` code if an exception is raised.

In this case, though, we have a `yield *inside* another with statement (`with tempdir(prefix=prefix) as tmpdir:`); AFAICT, I think that with statement (`with tempdir…`) will necessarily clean up the tempdir either way. Is that incorrect?